pbm/client.go

```diff
@@ -94,3 +95,170 @@ func (c *Client) CheckRequirements(ctx context.Context, hubs []types.PbmPlacemen
 
 	return res.Returnval, nil
 }
+
+func (c *Client) GetCompatibleDatastores(ctx context.Context, datastores []vim.ManagedObjectReference, ref *types.PbmServerObjectRef, ids []types.PbmProfileId) ([]vim.ManagedObjectReference, error) {
+	var hubs []types.PbmPlacementHub
+	var req []types.BasePbmPlacementRequirement
+	var compatibleDatastores []vim.ManagedObjectReference
+
+	for _, ds := range datastores {
+		hubs = append(hubs, types.PbmPlacementHub{
+			HubType: ds.Type,
+			HubId:   ds.Value,
+		})
+	}
+
+	for _, id := range ids {
+		req = append(req, &types.PbmPlacementCapabilityProfileRequirement{
+			ProfileId: id,
+		})
+	}
+
+	compatibilityResult, err := c.CheckRequirements(ctx, hubs, ref, req)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, res := range compatibilityResult {
+		if len(res.Error) == 0 {
+			compatibleDatastores = append(compatibleDatastores, vim.ManagedObjectReference{
+				Type:  res.Hub.HubType,
+				Value: res.Hub.HubId,
+			})
+		}
+	}
+
+	return compatibleDatastores, nil
+}
+
+func (c *Client) GetNonCompatibleDatastores(ctx context.Context, datastores []vim.ManagedObjectReference, ref *types.PbmServerObjectRef, ids []types.PbmProfileId) ([]vim.ManagedObjectReference, error) {
```

Comment:
Or single method + a struct that includes both to avoid making the remote call twice.

Another option, change `CheckRequirements` to return a helper type:
`type PlacementCompatibilityResult
[]types.PbmPlacementCompatibilityResult`

And attach helper methods to the helper type to filter/transform/etc in memory.

We do similar elsewhere, for example:
https://github.com/vmware/govmomi/blob/master/object/host_firewall_system.go#L82
https://github.com/vmware/govmomi/blob/master/object/authorization_manager.go#L40